src/combinatorics/simple_graph/basic.lean

```diff
@@ -292,6 +313,8 @@ A locally finite simple graph is regular of degree `d` if every vertex has degre
 -/
 def is_regular_of_degree (d : ℕ) : Prop := ∀ (v : V), G.degree v = d
 
+lemma is_regular_degree_eq {d : ℕ} (h : G.is_regular_of_degree d) (v : V) : G.degree v = d := h v
```

Comment:
The use case I had in mind when I suggested it was being able to `simp [G.is_regular_of_degree_eq h]` to rewrite every `G.degree` in an expression.  Otherwise you have to expand out the definition for `h` and then `simp` with it, I think.

Reply:
For a specific example, if we're rewriting inside of a sum, eg if I want to show the sum of degrees of a regular graph is something, then `simp [h v]` wouldn't be strong enough since it would only work for the specific `v`, but `simp [G.is_regular_of_degree_eq h]` would. (That said, maybe `sum_const_nat` would be better here, but for more complex expressions I think this still applies).

Reply:
Thanks, that makes sense!